app/controllers/api/v4/runs_controller.rb

```diff
@@ -14,11 +14,28 @@ class Api::V4::RunsController < Api::V4::ApplicationController
   end
 
   def show
-    if params[:historic] == '1'
-      @run.parse(fast: false)
-      render json: @run, serializer: Api::V4::RunWithHistorySerializer
+    accept = request.headers.fetch('HTTP_ACCEPT', 'json').downcase
+    timer = Run.program(accept)
+    json_out = accept == 'json' || timer.nil? || !Run.exportable_programs.include?(timer)
+    if json_out && accept != "original_timer"
+      if params[:historic] == '1'
+        @run.parse(fast: false)
+        render json: @run, serializer: Api::V4::RunWithHistorySerializer
+      else
+        render json: @run, serializer: Api::V4::RunSerializer
+      end
     else
-      render json: @run, serializer: Api::V4::RunSerializer
+      send_data(
+        if timer == Run.program(@run.timer) || accept == 'original_timer'
+          @run.file
+        else
+          render_to_string(file: "runs/#{accept}.html.erb", layout: false)
+        end,
+        layout: false,
+        type: "application/#{accept}",
```

Comment:
In case of Accept being `original_timer`, this would be `application/original_timer`. I think it may make sense to adjust it properly in that case. It's not important for us, but I think it makes more sense.

Reply:
That would make much more sense, you are quite right.